README.md

```diff
@@ -1,59 +1,85 @@
+[guard-gate](pkg/guard-gate) uses **Guardian** - a set of micro-rules that define the expected behavior of the service.
+
+The **Guardian** object maintains the following configuration parameters:
+
+1. A control object which defines the working mode of Guard in relation to the service
+1. A learned set of micro-rules updated by [guard-learner](cmd/guard-learner)
+1. A manually configured set of micro-rules that can be updated using `kubectl` or using [guard-ui](cmd/guard-ui)  
+
+Each service has its own respective **Guardian**. New services may use a default **Guardian** for the Kubernetes namespace or may use the default Guard **Guardian**. The following procedure describes how the **Guardian** for service `servicename` in namespace `namespace` is found. Note that [guard-gate](pkg/guard-gate) may be configured to either work with CRDs (guardians.wsecurity.ibmresearch.com) or with Configmaps `guardian-*` as **Guardians**.
+
+When looking for a service **Guardian**:
+
+- A **Dedicated-Guardian** per service is used (CRD named `<servicename>` or Configmap named `guardian-<servicename>`).
+- If the **Dedicated-Guardian** is not found, a **Namespace-Default -Guardian** is used as a starting point (CRD named `ns-<namespace>` or Configmap named `guardian-ns-<namespace>`).
+- If a **Namespace-Default-Guardian** is not found, a **Default-Guardian** is used as a starting point.
+- Note that the **Default-Guardian** uses auto-learning to help bootstrap new **Guardians**.
+
+See [guard-gate](pkg/guard-gate) for more details on the different Guard working modes.
+
+## Guard Learner
 
-Each service has its own respective *Guardian*. New services may use a default *Guardian* for the Kubernetes namespace or may use the default Guard *Guardian*. The following procedure describes how the *Guardian* for a service is found:
-- *Guardian* may reside in a CRD (guardians.wsecurity.ibmresearch.com) under the name '\<servicename\>.\<namespace\>' or in a configmap under the name 'guardian-\<servicename\>'. 
-- If a *Guardian* is not found, [`guard-gate`](https:pkg/guard-gate) will look for a namespace-default *Guardian* as a starting point under the name  '\<ns\>-\<namespace\>' or in a configmap under the name 'guardian-\<ns\>-\<namespace\>'.  
-- If a namespace-default *Guardian* is not found, [`guard-gate`](https:pkg/guard-gate) will use an empty set of micro-rules as a starting point and will set itself to work in auto-learning mode 
- 
-See [`guard-gate`](https:pkg/guard-gate) for more details on the different Guard working modes.
+[guard-learner](cmd/guard-learner) is a standalone service used to learn **Guardian** micro-rules based on inputs from instances of [guard-gate](pkg/guard-gate). [guard-learner](cmd/guard-learner) stores the **Guardian** as a CRD (guardians.wsecurity.ibmresearch.com) under the name `<servicename>` or in a Configmap under the name `guardian-<servicename>`.
```

Comment:
Is there a type in the IBM Research link?

Reply:
The guardians.wsecurity.ibmresearch.com is the CRD that is used by Guard:
```
kubectl get crds
NAME                                                  CREATED AT
...
certificates.networking.internal.knative.dev          2022-07-13T17:15:01Z
...
clusterdomainclaims.networking.internal.knative.dev   2022-07-13T17:15:03Z
...
configurations.serving.knative.dev                    2022-07-13T17:15:02Z
domainmappings.serving.knative.dev                    2022-07-13T17:15:03Z
...
guardians.wsecurity.ibmresearch.com                   2022-07-30T06:54:39Z
...
revisions.serving.knative.dev                         2022-07-13T17:15:05Z
routes.serving.knative.dev                            2022-07-13T17:15:06Z
serverlessservices.networking.internal.knative.dev    2022-07-13T17:15:06Z
services.serving.knative.dev                          2022-07-13T17:15:07Z
...
```

And than there is a CRD object to any monitored service (that is if CRDs are used and not Configmaps)
```
kubectl get guardians.wsecurity.ibmresearch.com
NAME            AGE
helloworld-go   10d
mytestservice   9d
```